IdentityCore/src/controllers/broker/ios/MSIDBrokerInteractiveController.m

```diff
@@ -178,119 +173,33 @@ - (void)acquireTokenImpl:(nonnull MSIDRequestCompletionBlock)completionBlock
-                                                                                              brokerKey:base64UrlKey
-                                                                                 brokerApplicationToken:applicationToken
-                                                                                        sdkCapabilities:self.sdkBrokerCapabilities
-                                                                                                  error:&brokerError];
-
-    if (!brokerRequest)
-    {
-        MSID_LOG_WITH_CTX(MSIDLogLevelError, self.requestParameters, @"Couldn't create broker request");
-        [self stopTelemetryEvent:[self telemetryAPIEvent] error:brokerError];
-        completionBlockWrapper(nil, brokerError);
-        return;
-    }
-    
-    NSDictionary *brokerResumeDictionary = brokerRequest.resumeDictionary;
-    [[NSUserDefaults standardUserDefaults] setObject:brokerResumeDictionary forKey:MSID_BROKER_RESUME_DICTIONARY_KEY];
-
-    [self callBrokerWithRequest:brokerRequest];
-}
-
-- (void)callBrokerWithRequest:(MSIDBrokerTokenRequest *)brokerRequest
-{
-    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Invoking broker for authentication, correlationId %@", brokerRequest.requestParameters.correlationId.UUIDString);
-    
-    [self.class setCurrentBrokerController:self];
-    [self.class startTrackingAppState];
-    [[MSIDTelemetry sharedInstance] startEvent:self.requestParameters.telemetryRequestId eventName:MSID_TELEMETRY_EVENT_LAUNCH_BROKER];
-
-    NSURL *brokerRequestURL = brokerRequest.brokerRequestURL;
-
-    NSURL *launchURL = _brokerInstallLink ? _brokerInstallLink : brokerRequestURL;
-    BOOL firstTimeInstall = _brokerInstallLink != nil;
-
-    if (firstTimeInstall)
-    {
-        [self saveToPasteBoard:brokerRequestURL];
-    }
-
-    if ([NSThread isMainThread])
-    {
-        [MSIDNotifications notifyWebAuthWillSwitchToBroker];
-        [self openBrokerWithRequestURL:launchURL fallbackToLocalController:!firstTimeInstall];
-    }
-    else
-    {
-        [MSIDMainThreadUtil executeOnMainThreadIfNeeded:^{
-            [MSIDNotifications notifyWebAuthWillSwitchToBroker];
-            [self openBrokerWithRequestURL:launchURL fallbackToLocalController:!firstTimeInstall];
-        }];
-    }
-}
-
-- (void)openBrokerWithRequestURL:(NSURL *)requestURL
-       fallbackToLocalController:(BOOL)shouldFallbackToLocalController
-{
-    NSDictionary *options = nil;
-    
-    if (self.interactiveParameters.brokerInvocationOptions.isUniversalLink)
-    {
-        // Option for openURL:options:CompletionHandler: only open URL if it is a valid universal link with an application configured to open it
-        // If there is no application configured, or the user disabled using it to open the link, completion handler called with NO
-        if (@available(iOS 10.0, *))
+    [MSIDBrokerUrlHandlerUtil UrlImplementation:self.brokerKeyProvider
```

Comment:
Could you provide more context why this particular change is needed?
Wouldn't MSAL CPP just use MSIDBrokerInteractiveController directly?
I believe we use it directly in MSAL Obj-C for both cases where we call broker directly.

Reply:
this class is gone